### DISCUSSION TITLE
How does the docker-compose.override.yml work?

### DISCUSSION THREAD

**Discussion by antiftw** [dunglas/symfony-docker#459] [Q&A] [answered]
Hi, 

I've just recently started migrating my entire setup to Docker so I'm rather new at all this. I've been rewriting the docker compose files since I'm using it to dockerize another symfony based project. Since I want to do it the right way, I'm trying to follow the setup used in this project, where there's one main file and other 'override' compose files to avoid duplicate code.

Now I was wondering how the `docker-compose.override.yml` works. I'm not really getting how Docker knows to use it, since in development the app is started simply with `docker compose up`. 

In production I understand that it knows to use `docker-compose.prod` since you call it using the two `-f` flags indicating the files that you want to use.

I've searched online and have found the `extends:` functionality - that I think is related to this subject - but I cannot seem to find that in any of the docker-compose files, so I'm kinda puzzled how this works.

Thanks in advance.

**Comment by maxhelias** [comment, marked as answer]
Hi

By default, Compose reads two files, a docker-compose.yml and an optional docker-compose.override.yml file. The override file can contain configuration overrides for existing services or entirely new services.
The [Extend](https://docs.docker.com/compose/multiple-compose-files/extends/) functionality allows you to extend a compose file by referring to another Compose file and selecting the bits you want to use in your own application, with the ability to override some attributes.

Doc here : https://docs.docker.com/compose/multiple-compose-files/

**Reply by antiftw** [reply]
Thanks for taking the time to respond. So it is just a default baked into Docker Compose. Docker could've mentioned that explicitly somewhere, because I had found the docs you mentioned but could not find that information on there.

Thanks again!
